frontend/src/scenes/insights/views/Funnels/FunnelStepsTable.tsx

```diff
@@ -72,18 +72,36 @@ export function FunnelStepsTable(): JSX.Element | null {
                         />
                     ),
                     dataIndex: 'breakdown_value',
-                    render: function RenderBreakdownValue(breakdownValue: BreakdownKeyType | undefined): JSX.Element {
-                        const label = formatBreakdownLabel(cohorts, formatPropertyValueForDisplay, breakdownValue)
+                    render: function RenderBreakdownValue(
+                        _: void,
+                        breakdown: FlattenedFunnelStepByBreakdown
+                    ): JSX.Element {
+                        // :KLUDGE: `BreakdownStepValues` is always wrapped into an array, which doesn't work for the
+                        // formatBreakdownLabel logic. Instead, we unwrap speculatively
+                        const value =
+                            breakdown.breakdown_value?.length == 1
+                                ? breakdown.breakdown_value[0]
+                                : breakdown.breakdown_value
+                        const label = formatBreakdownLabel(
+                            cohorts,
+                            formatPropertyValueForDisplay,
+                            value,
```

Comment:
The problem here isn't in this function - it's `getBreakdownStepValues` a couple of steps up from here. For some reason I wasn't able to discern it doesn't return data in `BreakdownKeyType` exactly - it _always_ wraps things into an array. This would cause formatBreakdownLabel not to function. Note that if we just passed the data from API to formatBreakdownLabel directly things would work.

Given this is specific to code touching `getBreakdownStepValues` (i.e. here) I am against making `formatBreakdownLabel` more general and instead we should expose the kludge here. Ideally also solve it but that's a much bigger refactoring.